lib/PipelineC/PipelineC.cpp

```diff
@@ -51,13 +53,18 @@ static bool loadLibraryPermanently(const char *LibraryPath) {
   return llvm::sys::DynamicLibrary::LoadLibraryPermanently(LibraryPath, &Msg);
 }
 
+static std::optional<revng::InitRevng> InitRevngInstance = std::nullopt;
+
 bool rp_initialize(int argc,
                    char *argv[],
                    int libraries_count,
                    const char *libraries_path[]) {
   if (Initialized)
     return false;
 
+  if (not InitRevngInstance.has_value())
```

Comment:
This is more like an `assert`, We don't want people to `rp_initialize`, `rp_shutdown`, `rp_initialize`, `rp_shutdown`...

Reply:
Why not? We only don't want them to do `rp_initialize`, `rp_initialize`. This is otherwise equivalent to

```cpp
void main() {
  {
    InitLLVM X(...);
  }
  {
    InitLLVM X(...);
  }
}
```

which is a bit dumb admittedly, but otherwise perfectly fine.

Reply:
You're trying to pull in a (more or less) reasonable feature without considering the context.
The line above makes it so that there's no point in doing what you're doing.
Let's introduce features only if we designed the whole system to have them, opportunistic introduction of features makes the code confusingly spaghetto.

Reply:
I changed it to `revng_check()`.